sw/host/tests/chip/gpio_intr/src/main.rs

```diff
@@ -0,0 +1,326 @@
+                PinmuxMioOut::Ior11 => PinmuxOutsel::GpioGpio26,
+                PinmuxMioOut::Ior12 => PinmuxOutsel::GpioGpio27,
+
+                PinmuxMioOut::Ior13 => PinmuxOutsel::GpioGpio28,
+            },
+        },
+    }
+});
+
+fn write_all_verify(
+    transport: &TransportWrapper,
+    uart: &dyn Uart,
+    value: u32,
+    config: &HashMap<PinmuxMioOut, PinmuxOutsel>,
+) -> Result<()> {
+    let mask = config.values().fold(0, |acc, v| {
+        let i = u32::from(*v) - u32::from(PinmuxOutsel::GpioGpio0);
+        acc | 1u32 << i
+    });
+    let masked_value = value & mask;
+    log::info!(
+        "write & verify pattern: {:08x} & {:08x} -> {:08x}",
+        value,
+        mask,
+        masked_value
+    );
+    if masked_value == 0 {
+        log::info!(
+            "skipping: {:08x} has no bits in common with the pinmux configuration",
+            value
+        );
+    } else {
+        GpioSet::write_all(uart, masked_value)?;
+        let mut result = 0u32;
+        for (k, v) in config.iter() {
+            let n = u32::from(transport.gpio_pin(&k.to_string())?.read()?);
+            let i = u32::from(*v) - u32::from(PinmuxOutsel::GpioGpio0);
+            result |= n << i;
+        }
+        log::info!("result = {:08x}", result);
+        assert_eq!(masked_value, result);
+        std::thread::sleep(Duration::from_secs(1));
+    }
+    Ok(())
+}
+
+//fn read_all_verify(
+//    transport: &TransportWrapper,
+//    uart: &dyn Uart,
+//    value: u32,
+//    config: &HashMap<PinmuxPeripheralIn, PinmuxInsel>,
+//) -> Result<()> {
+//    let mask = config.keys().fold(0, |acc, k| {
+//        let i = u32::from(*k) - u32::from(PinmuxPeripheralIn::GpioGpio0);
+//        acc | 1u32 << i
+//    });
+//    let masked_value = value & mask;
+//    log::info!(
+//        "read & verify pattern: {:08x} & {:08x} -> {:08x}",
+//        value,
+//        mask,
+//        masked_value
+//    );
+//    if masked_value == 0 {
+//        log::info!(
+//            "skipping: {:08x} has no bits in common with the pinmux configuration",
+//            value
+//        );
+//    } else {
+//        for (k, v) in config.iter() {
+//            let i = u32::from(*k) - u32::from(PinmuxPeripheralIn::GpioGpio0);
+//            transport
+//                .gpio_pin(&v.to_string())?
+//                .write((masked_value >> i) & 1 != 0)?;
+//        }
+//        let result = GpioGet::read_all(uart)? & mask;
+//        log::info!("result = {:08x}", result);
+//        assert_eq!(masked_value, result);
+//    }
+//    Ok(())
+//}
+
+fn test_gpio_outputs(opts: &Opts, transport: &TransportWrapper) -> Result<()> {
+    let uart = transport.uart("console")?;
+    log::info!(
+        "Configuring pinmux for {:?}",
+        opts.init.backend_opts.interface
+    );
+    let config = CONFIG
+        .get(opts.init.backend_opts.interface.as_str())
+        .expect("interface");
+    PinmuxConfig::configure(&*uart, None, Some(&config.output))?;
```

Comment:
If you want to do looped back / self-triggered interrupt stimulus, then the inputs will need to be connected too:

```suggestion
    PinmuxConfig::configure(&*uart, Some(&config.input), Some(&config.output))?;
```